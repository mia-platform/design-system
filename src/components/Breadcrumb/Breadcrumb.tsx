/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactElement, useMemo } from 'react'
import classNames from 'classnames'

import { BreadcrumbItem } from './BreadcrumbItem'
import { BreadcrumbProps } from './Breadcrumb.props'
import styles from './Breadcrumb.module.css'

const { breadcrumb } = styles

/**
 * UI component for displaying the current location within an hierarchy
 *
 * @returns {Breadcrumb} Breadcrumb component
 */
export const Breadcrumb = ({
  items,
}: BreadcrumbProps): ReactElement => {
  const breadcrumbClassNames = useMemo(() => classNames(
    [
      breadcrumb,
    ]
  ), [])

  return (
    <div className={breadcrumbClassNames}>
      {items.map(({ icon, menu, onClick, label }, index) =>
        <BreadcrumbItem
          icon={icon}
          index={index}
          itemsLength={items.length}
          key={'breadcrumb-item'}
          label={label}
          menu={menu}
          onClick={onClick}
        />
      )}
    </div>
  )
}
