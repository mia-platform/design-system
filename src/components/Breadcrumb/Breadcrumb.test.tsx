/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { fireEvent, render, screen, waitFor, waitForElementToBeRemoved } from '../../test-utils'
import { Breadcrumb } from '.'
import { BreadcrumbProps } from './Breadcrumb.props'
import { breadcrumbIcon } from './Breadcrumb.mocks'

describe('Breadcrumb Component', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  test('renders a breadcrumb with empty items', () => {
    const { asFragment } = render(<Breadcrumb items={[]} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders a breadcrumb with different button types', () => {
    const props: BreadcrumbProps = {
      items: [
        { },
        { label: 'Text' },
        { label: 'Very long text that should be ellipsed at some point' },
        { icon: breadcrumbIcon },
        { label: 'Text', icon: breadcrumbIcon },
        { menu: { items: [] } },
        { label: 'Text', menu: { items: [] } },
        { icon: breadcrumbIcon, label: 'Text', menu: { items: [] } },
        { onClick: jest.fn, icon: breadcrumbIcon, label: 'Text', menu: { items: [] } },
      ],
    }

    const { asFragment } = render(<Breadcrumb {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders a breadcrumb in loading state', () => {
    const props: BreadcrumbProps = {
      isLoading: true,
      items: [
        { label: 'Text' },
        { label: 'Text' },
        { label: 'Text' },
      ],
    }

    const { asFragment } = render(<Breadcrumb {...props} />)

    expect(asFragment()).toMatchSnapshot()

    // Fire a resize to trigger hidden items computation
    fireEvent(window, new Event('resize'))
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders item menu correctly', async() => {
    const buttonClickMock = jest.fn()
    const menuItemClickMock = jest.fn()
    const onDropdownVisibleChangeMock = jest.fn()

    const props: BreadcrumbProps = {
      items: [
        {
          key: 'root',
          onClick: buttonClickMock,
          label: 'Text',
          icon: breadcrumbIcon,
          menu: {
            onClick: menuItemClickMock,
            onDropdownVisibleChange: onDropdownVisibleChangeMock,
            search: true,
            items: [
              { key: 'item-1', label: 'Item 1' },
              { key: 'item-2', label: 'Item 2', icon: breadcrumbIcon },
            ],
          },
        },
        {
          menu: {
            activeKey: 'sub-item',
            items: [{ key: 'sub-item', label: 'Sub item', icon: breadcrumbIcon }],
          },
        },
      ],
      getPopupContainer: (containerNode) => containerNode,
    }

    const { asFragment } = render(<Breadcrumb {...props} />)
    expect(asFragment()).toMatchSnapshot()

    const [button] = screen.getAllByText('Text')
    fireEvent.click(button)
    expect(buttonClickMock).toHaveBeenCalledTimes(1)

    const [dropdownTrigger1, dropdownTrigger2] = screen.getAllByLabelText('caret-full-down')

    fireEvent.click(dropdownTrigger1)
    expect(buttonClickMock).toHaveBeenCalledTimes(1)
    expect(onDropdownVisibleChangeMock).toHaveBeenCalledTimes(1)
    expect(onDropdownVisibleChangeMock).toHaveBeenCalledWith(true)

    expect(asFragment()).toMatchSnapshot()

    fireEvent.click(screen.getByText('Item 1'))
    expect(menuItemClickMock).toHaveBeenCalledTimes(1)
    expect(menuItemClickMock).toHaveBeenCalledWith('item-1', expect.any(Object))

    fireEvent.click(dropdownTrigger1)
    fireEvent.click(screen.getByText('Item 2'))
    expect(menuItemClickMock).toHaveBeenNthCalledWith(2, 'item-2', expect.any(Object))

    fireEvent.click(dropdownTrigger1)
    fireEvent.change(screen.getByPlaceholderText('Search...'), { target: { value: 'foo' } })
    await waitFor(() => expect(screen.getByText('No items')).toBeInTheDocument())

    fireEvent.click(dropdownTrigger1)
    expect(screen.queryByText('No items')).not.toBeInTheDocument()

    fireEvent.click(dropdownTrigger1)
    expect(screen.queryByText('No items')).not.toBeInTheDocument()
    expect(screen.getByText('Item 1')).toBeInTheDocument()
    expect(screen.getByText('Item 2')).toBeInTheDocument()
    fireEvent.change(screen.getByPlaceholderText('Search...'), { target: { value: '1' } })
    await waitFor(() => expect(screen.getByText('Item 1')).toBeInTheDocument())
    await waitForElementToBeRemoved(() => screen.queryByText('Item 2'))

    fireEvent.click(dropdownTrigger2)
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders controlled item menu correctly', () => {
    const onSearchMock = jest.fn()
    const menuItemClickMock = jest.fn()
    const onDropdownVisibleChangeMock = jest.fn()

    const props: BreadcrumbProps = {
      items: [
        {
          key: 'root',
          label: 'Text',
          icon: breadcrumbIcon,
          menu: {
            isOpen: true,
            onClick: menuItemClickMock,
            onDropdownVisibleChange: onDropdownVisibleChangeMock,
            search: {
              onChange: onSearchMock,
            },
            items: [
              { key: 'item-1', label: 'Item 1' },
              { key: 'item-2', label: 'Item 2', icon: breadcrumbIcon },
            ],
          },
        },
      ],
    }

    const { asFragment } = render(<Breadcrumb {...props} />)
    expect(asFragment()).toMatchSnapshot()

    expect(screen.getByText('Item 1')).toBeInTheDocument()
    expect(screen.getByText('Item 2')).toBeInTheDocument()

    const [button] = screen.getAllByText('Text')
    fireEvent.click(button)
    expect(onDropdownVisibleChangeMock).toHaveBeenNthCalledWith(1, false)
    expect(screen.getByText('Item 1')).toBeInTheDocument()
    expect(screen.getByText('Item 2')).toBeInTheDocument()

    fireEvent.change(screen.getByPlaceholderText('Search...'), { target: { value: 'foo' } })
    expect(onSearchMock).toHaveBeenNthCalledWith(1, expect.any(Object))
    expect(screen.getByText('Item 1')).toBeInTheDocument()
    expect(screen.getByText('Item 2')).toBeInTheDocument()
  })

  test('renders a breadcrumb with collapsed items', () => {
    const onClickMock = jest.fn()
    const onDropdownVisibleChangeMock = jest.fn()

    const props: BreadcrumbProps = {
      items: [
        { label: 'Text 1' },
        { label: 'Text 2' },
        { label: 'Text 3', onClick: jest.fn },
        { label: 'Text 4', icon: breadcrumbIcon },
        { label: 'Text 5', menu: { items: [{ label: 'Item 1' }] } },
        {
          menu: {
            activeKey: '1',
            items: [{ key: '1', label: 'Item 1', icon: breadcrumbIcon }],
          },
        },
        { label: 'Text 7', onClick: onClickMock },
        {
          label: 'Text 8',
          menu: {
            onDropdownVisibleChange: onDropdownVisibleChangeMock,
            items: [{ label: 'Item 2' }],
          },
        },
      ],
    }

    const { asFragment } = render(<Breadcrumb {...props} />)

    // Fire a resize to trigger hidden items computation
    fireEvent(window, new Event('resize'))

    expect(asFragment()).toMatchSnapshot()

    expect(screen.getAllByText('Text 2')).toHaveLength(1)

    const collapseButton = screen.getByLabelText('PiDotsThree')
    fireEvent.click(collapseButton)

    expect(asFragment()).toMatchSnapshot()

    expect(screen.getAllByText('Text 2')).toHaveLength(2)

    fireEvent.click(screen.getAllByText('Text 7')[1])
    expect(onClickMock).toHaveBeenCalledTimes(1)

    fireEvent.click(screen.getAllByText('Text 7')[0])
    expect(onClickMock).toHaveBeenCalledTimes(1)

    fireEvent.click(screen.getAllByText('Text 8')[0])
    expect(onDropdownVisibleChangeMock).toHaveBeenCalledTimes(1)

    fireEvent.click(screen.getAllByText('Text 8')[1])
    expect(onDropdownVisibleChangeMock).toHaveBeenCalledTimes(1)
  })

  test('should get breadcrumb by test id', () => {
    render(<Breadcrumb dataTestId="breadcrumb-test" items={[]} />)
    expect(screen.getByTestId('breadcrumb-test')).toBeTruthy()
  })
})
