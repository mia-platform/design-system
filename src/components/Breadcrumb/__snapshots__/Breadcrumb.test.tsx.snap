// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb Component renders a breadcrumb with multiple items - with menu 1`] = `
<DocumentFragment>
  <div
    class="breadcrumb"
  >
    <div
      class="breadcrumbItemWrapper"
    >
      <div
        class="breadcrumbItemLabelWrapper"
      >
        <svg
          alt="PiCircleHalfTiltLight"
          aria-label="PiCircleHalfTiltLight"
          color="black"
          fill="currentColor"
          height="16"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="color: black;"
          viewBox="0 0 256 256"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M200.12,55.88A102,102,0,1,0,55.88,200.13,102,102,0,1,0,200.12,55.88ZM90,209.62a89.61,89.61,0,0,1-21.23-13.89L90,174.49Zm32,8.16a90,90,0,0,1-20-3.58V162.49l20-20Zm32-3.58a89.8,89.8,0,0,1-20,3.58V130.49l20-20Zm32-17.4a89.45,89.45,0,0,1-20,12.83V98.49l20-20ZM60.27,187.24a90,90,0,0,1,127-127ZM198,184.57V71.43a90,90,0,0,1,0,113.14Z"
          />
        </svg>
        <div
          class="breadcrumbItemLabelStyle"
        >
          <div
            class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-single-line mia-platform-typography-ellipsis-single-line bodyL"
            role="paragraph"
          >
            first
          </div>
        </div>
      </div>
      <div
        class="mia-platform-dropdown-trigger breadcrumbMenuIcon"
      >
        <svg
          alt="AiOutlineCaretDown"
          aria-label="AiOutlineCaretDown"
          color="#636363"
          fill="currentColor"
          height="16"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(99, 99, 99);"
          viewBox="0 0 1024 1024"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
          />
        </svg>
      </div>
    </div>
    <svg
      alt="PiCaretRight"
      aria-label="PiCaretRight"
      color="#636363"
      fill="currentColor"
      height="16"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(99, 99, 99);"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
      />
    </svg>
    <div
      class="breadcrumbItemWrapper"
    >
      <div
        class="breadcrumbItemLabelWrapper withoutMenu"
      >
        <svg
          alt="PiCircleHalfTiltLight"
          aria-label="PiCircleHalfTiltLight"
          color="black"
          fill="currentColor"
          height="16"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="color: black;"
          viewBox="0 0 256 256"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M200.12,55.88A102,102,0,1,0,55.88,200.13,102,102,0,1,0,200.12,55.88ZM90,209.62a89.61,89.61,0,0,1-21.23-13.89L90,174.49Zm32,8.16a90,90,0,0,1-20-3.58V162.49l20-20Zm32-3.58a89.8,89.8,0,0,1-20,3.58V130.49l20-20Zm32-17.4a89.45,89.45,0,0,1-20,12.83V98.49l20-20ZM60.27,187.24a90,90,0,0,1,127-127ZM198,184.57V71.43a90,90,0,0,1,0,113.14Z"
          />
        </svg>
        <div
          class="breadcrumbItemLabelStyle"
        >
          <div
            class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-single-line mia-platform-typography-ellipsis-single-line bodyL"
            role="paragraph"
          >
            empty items
          </div>
        </div>
      </div>
    </div>
    <svg
      alt="PiCaretRight"
      aria-label="PiCaretRight"
      color="#636363"
      fill="currentColor"
      height="16"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(99, 99, 99);"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
      />
    </svg>
    <div
      class="breadcrumbItemWrapper"
    >
      <div
        class="breadcrumbItemLabelWrapper"
      >
        <svg
          alt="PiCircleHalfTiltLight"
          aria-label="PiCircleHalfTiltLight"
          color="black"
          fill="currentColor"
          height="16"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="color: black;"
          viewBox="0 0 256 256"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M200.12,55.88A102,102,0,1,0,55.88,200.13,102,102,0,1,0,200.12,55.88ZM90,209.62a89.61,89.61,0,0,1-21.23-13.89L90,174.49Zm32,8.16a90,90,0,0,1-20-3.58V162.49l20-20Zm32-3.58a89.8,89.8,0,0,1-20,3.58V130.49l20-20Zm32-17.4a89.45,89.45,0,0,1-20,12.83V98.49l20-20ZM60.27,187.24a90,90,0,0,1,127-127ZM198,184.57V71.43a90,90,0,0,1,0,113.14Z"
          />
        </svg>
        <div
          class="breadcrumbItemLabelStyle"
        >
          <div
            class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-single-line mia-platform-typography-ellipsis-single-line bodyL"
            role="paragraph"
          >
            with search
          </div>
        </div>
      </div>
      <div
        class="mia-platform-dropdown-trigger breadcrumbMenuIcon"
      >
        <svg
          alt="AiOutlineCaretDown"
          aria-label="AiOutlineCaretDown"
          color="#636363"
          fill="currentColor"
          height="16"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(99, 99, 99);"
          viewBox="0 0 1024 1024"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
          />
        </svg>
      </div>
    </div>
    <svg
      alt="PiCaretRight"
      aria-label="PiCaretRight"
      color="#636363"
      fill="currentColor"
      height="16"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(99, 99, 99);"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
      />
    </svg>
    <div
      class="breadcrumbItemWrapper"
    >
      <div
        class="mia-platform-dropdown-trigger breadcrumbMenuIcon caretOnly"
      >
        <svg
          alt="AiOutlineCaretDown"
          aria-label="AiOutlineCaretDown"
          color="#636363"
          fill="currentColor"
          height="16"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(99, 99, 99);"
          viewBox="0 0 1024 1024"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
          />
        </svg>
      </div>
    </div>
    <svg
      alt="PiCaretRight"
      aria-label="PiCaretRight"
      color="#636363"
      fill="currentColor"
      height="16"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(99, 99, 99);"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
      />
    </svg>
    <div
      class="breadcrumbItemWrapper"
    >
      <div
        class="breadcrumbItemLabelWrapper last"
      >
        <div
          class="breadcrumbItemLabelStyle"
        >
          <div
            class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-single-line mia-platform-typography-ellipsis-single-line bodyL"
            role="paragraph"
          >
            without search
          </div>
        </div>
      </div>
      <div
        class="mia-platform-dropdown-trigger breadcrumbMenuIcon"
      >
        <svg
          alt="AiOutlineCaretDown"
          aria-label="AiOutlineCaretDown"
          color="#636363"
          fill="currentColor"
          height="16"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(99, 99, 99);"
          viewBox="0 0 1024 1024"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Breadcrumb Component renders a breadcrumb with multiple items - without menu (Default) 1`] = `
<DocumentFragment>
  <div
    class="breadcrumb"
  >
    <div
      class="breadcrumbItemWrapper"
    >
      <div
        class="breadcrumbItemLabelWrapper initial withoutMenu"
      >
        <svg
          alt="PiCircleHalfTiltLight"
          aria-label="PiCircleHalfTiltLight"
          color="black"
          fill="currentColor"
          height="16"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="color: black;"
          viewBox="0 0 256 256"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M200.12,55.88A102,102,0,1,0,55.88,200.13,102,102,0,1,0,200.12,55.88ZM90,209.62a89.61,89.61,0,0,1-21.23-13.89L90,174.49Zm32,8.16a90,90,0,0,1-20-3.58V162.49l20-20Zm32-3.58a89.8,89.8,0,0,1-20,3.58V130.49l20-20Zm32-17.4a89.45,89.45,0,0,1-20,12.83V98.49l20-20ZM60.27,187.24a90,90,0,0,1,127-127ZM198,184.57V71.43a90,90,0,0,1,0,113.14Z"
          />
        </svg>
        <div
          class="breadcrumbItemLabelStyle"
        >
          <div
            class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-single-line mia-platform-typography-ellipsis-single-line bodyL"
            role="paragraph"
          >
            first
          </div>
        </div>
      </div>
    </div>
    <svg
      alt="PiCaretRight"
      aria-label="PiCaretRight"
      color="#636363"
      fill="currentColor"
      height="16"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(99, 99, 99);"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
      />
    </svg>
    <div
      class="breadcrumbItemWrapper"
    >
      <div
        class="breadcrumbItemLabelWrapper withoutMenu"
      >
        <svg
          alt="PiCircleHalfTiltLight"
          aria-label="PiCircleHalfTiltLight"
          color="black"
          fill="currentColor"
          height="16"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="color: black;"
          viewBox="0 0 256 256"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M200.12,55.88A102,102,0,1,0,55.88,200.13,102,102,0,1,0,200.12,55.88ZM90,209.62a89.61,89.61,0,0,1-21.23-13.89L90,174.49Zm32,8.16a90,90,0,0,1-20-3.58V162.49l20-20Zm32-3.58a89.8,89.8,0,0,1-20,3.58V130.49l20-20Zm32-17.4a89.45,89.45,0,0,1-20,12.83V98.49l20-20ZM60.27,187.24a90,90,0,0,1,127-127ZM198,184.57V71.43a90,90,0,0,1,0,113.14Z"
          />
        </svg>
        <div
          class="breadcrumbItemLabelStyle"
        >
          <div
            class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-single-line mia-platform-typography-ellipsis-single-line bodyL"
            role="paragraph"
          >
            second
          </div>
        </div>
      </div>
    </div>
    <svg
      alt="PiCaretRight"
      aria-label="PiCaretRight"
      color="#636363"
      fill="currentColor"
      height="16"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(99, 99, 99);"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
      />
    </svg>
    <div
      class="breadcrumbItemWrapper"
    >
      <div
        class="breadcrumbItemLabelWrapper withoutMenu"
      >
        <div
          class="breadcrumbItemLabelStyle"
        >
          <div
            class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-single-line mia-platform-typography-ellipsis-single-line bodyL"
            role="paragraph"
          >
            third
          </div>
        </div>
      </div>
    </div>
    <svg
      alt="PiCaretRight"
      aria-label="PiCaretRight"
      color="#636363"
      fill="currentColor"
      height="16"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(99, 99, 99);"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
      />
    </svg>
    <div
      class="breadcrumbItemWrapper"
    >
      <div
        class="breadcrumbItemLabelWrapper last withoutMenu"
      >
        <svg
          alt="PiCircleHalfTiltLight"
          aria-label="PiCircleHalfTiltLight"
          color="black"
          fill="currentColor"
          height="16"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="color: black;"
          viewBox="0 0 256 256"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M200.12,55.88A102,102,0,1,0,55.88,200.13,102,102,0,1,0,200.12,55.88ZM90,209.62a89.61,89.61,0,0,1-21.23-13.89L90,174.49Zm32,8.16a90,90,0,0,1-20-3.58V162.49l20-20Zm32-3.58a89.8,89.8,0,0,1-20,3.58V130.49l20-20Zm32-17.4a89.45,89.45,0,0,1-20,12.83V98.49l20-20ZM60.27,187.24a90,90,0,0,1,127-127ZM198,184.57V71.43a90,90,0,0,1,0,113.14Z"
          />
        </svg>
        <div
          class="breadcrumbItemLabelStyle"
        >
          <div
            class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-single-line mia-platform-typography-ellipsis-single-line bodyL"
            role="paragraph"
          >
            fourth
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Breadcrumb Component renders a breadcrumb with one item - icon and title 1`] = `
<DocumentFragment>
  <div
    class="breadcrumb"
  >
    <div
      class="breadcrumbItemWrapper"
    >
      <div
        class="breadcrumbItemLabelWrapper initial last withoutMenu"
      >
        <svg
          alt="PiCircleHalfTiltLight"
          aria-label="PiCircleHalfTiltLight"
          color="black"
          fill="currentColor"
          height="16"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="color: black;"
          viewBox="0 0 256 256"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M200.12,55.88A102,102,0,1,0,55.88,200.13,102,102,0,1,0,200.12,55.88ZM90,209.62a89.61,89.61,0,0,1-21.23-13.89L90,174.49Zm32,8.16a90,90,0,0,1-20-3.58V162.49l20-20Zm32-3.58a89.8,89.8,0,0,1-20,3.58V130.49l20-20Zm32-17.4a89.45,89.45,0,0,1-20,12.83V98.49l20-20ZM60.27,187.24a90,90,0,0,1,127-127ZM198,184.57V71.43a90,90,0,0,1,0,113.14Z"
          />
        </svg>
        <div
          class="breadcrumbItemLabelStyle"
        >
          <div
            class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-single-line mia-platform-typography-ellipsis-single-line bodyL"
            role="paragraph"
          >
            Text
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Breadcrumb Component renders a breadcrumb with one item - title only 1`] = `
<DocumentFragment>
  <div
    class="breadcrumb"
  >
    <div
      class="breadcrumbItemWrapper"
    >
      <div
        class="breadcrumbItemLabelWrapper initial last withoutMenu"
      >
        <div
          class="breadcrumbItemLabelStyle"
        >
          <div
            class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-single-line mia-platform-typography-ellipsis-single-line bodyL"
            role="paragraph"
          >
            Text
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Breadcrumb Component renders a breadcrumb with two items 1`] = `
<DocumentFragment>
  <div
    class="breadcrumb"
  >
    <div
      class="breadcrumbItemWrapper"
    >
      <div
        class="breadcrumbItemLabelWrapper initial withoutMenu"
      >
        <svg
          alt="PiCircleHalfTiltLight"
          aria-label="PiCircleHalfTiltLight"
          color="black"
          fill="currentColor"
          height="16"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="color: black;"
          viewBox="0 0 256 256"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M200.12,55.88A102,102,0,1,0,55.88,200.13,102,102,0,1,0,200.12,55.88ZM90,209.62a89.61,89.61,0,0,1-21.23-13.89L90,174.49Zm32,8.16a90,90,0,0,1-20-3.58V162.49l20-20Zm32-3.58a89.8,89.8,0,0,1-20,3.58V130.49l20-20Zm32-17.4a89.45,89.45,0,0,1-20,12.83V98.49l20-20ZM60.27,187.24a90,90,0,0,1,127-127ZM198,184.57V71.43a90,90,0,0,1,0,113.14Z"
          />
        </svg>
        <div
          class="breadcrumbItemLabelStyle"
        >
          <div
            class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-single-line mia-platform-typography-ellipsis-single-line bodyL"
            role="paragraph"
          >
            first
          </div>
        </div>
      </div>
    </div>
    <svg
      alt="PiCaretRight"
      aria-label="PiCaretRight"
      color="#636363"
      fill="currentColor"
      height="16"
      role="img"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(99, 99, 99);"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
      />
    </svg>
    <div
      class="breadcrumbItemWrapper"
    >
      <div
        class="breadcrumbItemLabelWrapper last withoutMenu"
      >
        <svg
          alt="PiCircleHalfTiltLight"
          aria-label="PiCircleHalfTiltLight"
          color="black"
          fill="currentColor"
          height="16"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="color: black;"
          viewBox="0 0 256 256"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M200.12,55.88A102,102,0,1,0,55.88,200.13,102,102,0,1,0,200.12,55.88ZM90,209.62a89.61,89.61,0,0,1-21.23-13.89L90,174.49Zm32,8.16a90,90,0,0,1-20-3.58V162.49l20-20Zm32-3.58a89.8,89.8,0,0,1-20,3.58V130.49l20-20Zm32-17.4a89.45,89.45,0,0,1-20,12.83V98.49l20-20ZM60.27,187.24a90,90,0,0,1,127-127ZM198,184.57V71.43a90,90,0,0,1,0,113.14Z"
          />
        </svg>
        <div
          class="breadcrumbItemLabelStyle"
        >
          <div
            class="mia-platform-typography mia-platform-typography-ellipsis mia-platform-typography-single-line mia-platform-typography-ellipsis-single-line bodyL"
            role="paragraph"
          >
            second
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
